front-end-web/src/Components/Pages/ProfilePage.tsx

```diff
@@ -0,0 +1,184 @@
+        type: "success",
+        open: true
+      });
+    }
+  };
+  const [openPass, setOpenPass] = React.useState(false);
+  const handleOpenPasswordDia = () => {
+    handleToggleProfile();
+    setOpenPass(true);
+  };
+  const handleClosePasswordDia = () => {
+    setOpenPass(false);
+  };
+
+  const handleNewPassword = () => {
+    if (newPassword === newPasswordCheck) {
+      handleChangePass();
+      handleClosePasswordDia();
+    } else {
+      handleNewNotification({
+        message: "Passwords do not match.",
+        type: "error",
+        open: true
+      });
+    }
+  };
+
+  const handleChangeName = () => {
+    const currentUser = firebase.auth().currentUser;
+    if (currentUser) {
+      currentUser.updateProfile({ displayName: newName });
+      handleNewNotification({
+        message: "Display name changed successfully.",
+        type: "success",
+        open: true
+      });
+    }
+  };
+  const currentUser = firebase.auth().currentUser;
+  if (currentUser) {
+    if (currentUser.displayName !== null) {
+      currentName = currentUser.displayName;
+    }
+    if (currentUser.email !== null) {
+      currentEmail = currentUser.email;
+    }
+  }
+  return (
+    <Fragment>
+      <Dialog open={openPage} aria-labelledby="form-dialog-title">
+        <DialogTitle id="form-dialog-title">User Profile</DialogTitle>
+        <DialogContent>
+          <DialogContentText></DialogContentText>
+          <TextField
+            margin="dense"
+            helperText="Display Name"
+            id="name"
+            fullWidth
+            defaultValue={currentName}
+            onChange={e => (newName = e.target.value)}
+          />
+        </DialogContent>
+        <Button onClick={handleChangeName} color="primary">
+          Save new name
+        </Button>
+        <TextField
+          margin="dense"
+          helperText="Email"
+          id="email"
+          fullWidth
+          type="email"
+          defaultValue={currentEmail}
+          onChange={e => (newEmail = e.target.value)}
+        />
+        <Button onClick={handleChangeEmail} color="primary">
+          Save new email
+        </Button>
+        <DialogActions>
+          <Button onClick={handleOpenPasswordDia} color="primary">
+            Reset Password
+          </Button>
+          <Button onClick={handleToggleProfile} color="primary">
+            Close
+          </Button>
+        </DialogActions>
+      </Dialog>
+      <Dialog open={openPass} aria-labelledby="form-dialog-title">
+        <DialogTitle id="form-dialog-title">Reset Password</DialogTitle>
+        <DialogContent>
+          <DialogContentText></DialogContentText>
+          <TextField
+            margin="dense"
+            helperText="Type a new password"
+            id="passMain"
+            fullWidth
+            type="password"
+            onChange={e => (newPassword = e.target.value)}
+          />
+        </DialogContent>
```

Comment:
`DialogContent` should include all of the content, including the buttons and email reset. I would move the closing tag until right before `DialogActions` is opened.